src/qml/chainmodel.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_QML_CHAINMODEL_H
+#define BITCOIN_QML_CHAINMODEL_H
+
+#include <interfaces/chain.h>
+
+#include <QObject>
+#include <QTimer>
+#include <QVariant>
+
+namespace interfaces {
+class FoundBlock;
+class Chain;
+} // namespace interfaces
+
+static const int SECS_IN_12_HOURS = 43200;
+
+class ChainModel : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QVariantList timeRatioList READ timeRatioList NOTIFY timeRatioListChanged)
+
+public:
+    explicit ChainModel(interfaces::Chain& chain);
+
+    QVariantList timeRatioList() const { return m_time_ratio_list; };
+
+    int timestampAtMeridian();
+
+    void setCurrentTimeRatio();
+
+public Q_SLOTS:
+    void setTimeRatioList(int new_time);
+    void setTimeRatioListInitial();
+
+Q_SIGNALS:
+    void timeRatioListChanged();
+
+private:
+    QVariantList m_time_ratio_list{0.0};
+
+    QTimer* timer;
```

Comment:
a287f9e44660b80349defa8f34f14c8b9f0d8b78
Naming convention:
```suggestion
    QTimer* m_timer;
```

Reply:
Addressed in 67d12a67536b636bbda3f5bb8d1e62cc5fa66d65

Since this timer didn't need to be a global variable, I changed it to a local variable.